src/FillArrays.jl

```diff
@@ -207,9 +207,11 @@ function +(a::Fill{T, 1}, b::AbstractRange) where {T}
     return a.value .+ b
 end
 +(a::AbstractRange, b::AbstractFill) = b + a
++(a::AbstractFill{<:Any,2}, b::UniformScaling) = Array(a) + b
```

Comment:
The reverse order `+(::UniformScaling, ::AbstractFill{<:Any,2})` falls back to this definition: https://github.com/JuliaLang/julia/blob/e3b681c6872d87d74db63dd7ec3da2beee8e3ae4/stdlib/LinearAlgebra/src/uniformscaling.jl#L153

Reply:
Not every `AbstractFill` can be converted to an `Array`, e.g. infinite arrays:
```julia
julia> using InfiniteArrays

julia> Fill(1,∞,∞) + Eye{Int}(∞)
(ℵ₀×ℵ₀ Fill{Int64, 2, Tuple{InfiniteArrays.OneToInf{Int64}, InfiniteArrays.OneToInf{Int64}}} with indices OneToInf()×OneToInf()) .+ (ℵ₀×ℵ₀ Diagonal{Int64, Ones{Int64, 1, Tuple{InfiniteArrays.OneToInf{Int64}}}} with indices OneToInf()×OneToInf()) with indices OneToInf()×OneToInf():
 2  1  1  1  1  1  1  1  1  1  1  1  1  …  
 1  2  1  1  1  1  1  1  1  1  1  1  1     
 1  1  2  1  1  1  1  1  1  1  1  1  1     
 1  1  1  2  1  1  1  1  1  1  1  1  1     
 1  1  1  1  2  1  1  1  1  1  1  1  1     
 1  1  1  1  1  2  1  1  1  1  1  1  1  …  
 1  1  1  1  1  1  2  1  1  1  1  1  1     
 1  1  1  1  1  1  1  2  1  1  1  1  1     
 1  1  1  1  1  1  1  1  2  1  1  1  1     
 1  1  1  1  1  1  1  1  1  2  1  1  1     
 1  1  1  1  1  1  1  1  1  1  2  1  1  …  
 1  1  1  1  1  1  1  1  1  1  1  2  1     
 1  1  1  1  1  1  1  1  1  1  1  1  2     
 1  1  1  1  1  1  1  1  1  1  1  1  1     
 1  1  1  1  1  1  1  1  1  1  1  1  1     
 1  1  1  1  1  1  1  1  1  1  1  1  1  …  
 1  1  1  1  1  1  1  1  1  1  1  1  1     
 1  1  1  1  1  1  1  1  1  1  1  1  1     
 ⋮              ⋮              ⋮        ⋱  
```
So:
```suggestion
+(a::AbstractFill{<:Any,2}, b::UniformScaling) = a + Diagonal(Fill(b.λ,(axes(a,1),)))
```

Reply:
Ah that's a good suggestion, I didn't consider arrays for which `Array` is not defined.